virtual_rainforest/models/abiotic/abiotic_model.py

```diff
@@ -66,22 +69,23 @@ def __init__(
             LOGGER.error(to_raise)
             raise to_raise
 
-        super().__init__(update_interval, start_time, **kwargs)
+        super().__init__(data, update_interval, start_time, **kwargs)
         self.soil_layers = int(soil_layers)
         self.canopy_layers = int(canopy_layers)
         # Save variables names to be used by the __repr__
         self._repr.append("soil_layers")
         self._repr.append("canopy_layers")
 
     @classmethod
-    def from_config(cls, config: dict[str, Any]) -> AbioticModel:
+    def from_config(cls, data: Data, config: dict[str, Any]) -> AbioticModel:
         """Factory function to initialise the abiotic model from configuration.
 
         This function unpacks the relevant information from the configuration file, and
         then uses it to initialise the model. If any information from the config is
         invalid rather than returning an initialised model instance None is returned.
 
         Args:
+            data: A :class:`~virtual_rainforest.core.data.Data` instance.
```

Comment:
Minor nit: the user will already be able to see `data`'s type, so there's no point in repeating it. How about describing it as "loaded user data" or something?